docs/guides/tfhers.md

```diff
@@ -0,0 +1,242 @@
+let server_key = client_key.generate_server_key();
+
+```
+
+#### KeyGen starts in TFHE-rs
+
+First, we generate the TFHE-rs keyset and then serialize the shared secret key that will be used to encrypt the inputs
+
+```rust
+
+use tfhe::{prelude::*, ConfigBuilder};
+use tfhe::generate_keys;
+
+/// ...
+
+// Concrete uses this parameters to define the TFHE-rs ciphertext type
+let config = ConfigBuilder::with_custom_parameters(
+        tfhe::shortint::prelude::PARAM_MESSAGE_2_CARRY_3_KS_PBS,
+    )
+.build();
+
+let (client_key, server_key) = generate_keys(config);
+let (integer_ck, _, _) = client_key.clone().into_raw_parts();
+let shortint_ck = integer_ck.into_raw_parts();
+let (glwe_secret_key, _, _) = shortint_ck.into_raw_parts();
+let lwe_secret_key = glwe_secret_key.into_lwe_secret_key();
+
+save_lwe_sk(lwe_secret_key, "secret_key_from_tfhers");
+
+```
+
+Next, we generate a Concrete keyset using the shared secret key from TFHE-rs.
+
+```python
+# this was generated from TFHE-rs
+with open("secret_key_from_tfhers", "rb") as f:
+    sk_buff = f.read()
+# maps input indices to their secret key
+input_idx_to_key = {0: sk_buff, 1: sk_buff}
+# we do a Concrete keygen starting with an initial set of secret keys
+tfhers_bridge.keygen_with_initial_keys(input_idx_to_key_buffer=input_idx_to_key)
+```
+
+### Using Ciphertexts
+
+We should have everything now to encrypt, compute and decrypt on both sides. We will now do an encryption and computation in TFHE-rs, move to Concrete, run our circuit, and come back to TFHE-rs for decryption.
+
+First, we do encryption and a simple addition in TFHE-rs. For more information on how to save ciphertexts, refer to [Serialization](#serialization-of-ciphertexts-and-keys).
+
+```rust
+let x = FheUint8::encrypt(20, &client_key);
+let y = FheUint8::encrypt(24, &client_key);
+let z = FheUint8::encrypt(9, &client_key);
+y += z;
+
+save_fheuint8(x, "tfhers_x");
+save_fheuint8(y, "tfhers_y");
+```
+
+Next, we can load these ciphertexts in Concrete and then run our compiled circuit as usual.
+
+```python
+with open("tfhers_x", "rb") as f:
+    buff_x = f.read()
+with open("tfhers_y", "rb") as f:
+    buff_y = f.read()
+tfhers_uint8_x = tfhers_bridge.import_value(buff_x, input_idx=0)
+tfhers_uint8_y = tfhers_bridge.import_value(buff_y, input_idx=1)
+
+encrypted_result = circuit.run(tfhers_uint8_x, tfhers_uint8_y)
+```
+
+Finally, we can decrypt and decode in Concrete ...
+
+```python
+result = circuit.decrypt(encrypted_result)
+decoded = tfhers_type.decode(result)
+```
+
+... or export it to TFHE-rs for computation/decryption
+
+```python
+buff_out = tfhers_bridge.export_value(encrypted_result, output_idx=0)
+# write it to file
+with open("tfhers_out", "wb") as f:
+    f.write(buff_out)
+```
+
+```rust
+
+let fheuint = load_fheuint8("tfhers_out");
+// you can do computation before decryption as well
+let result: u8 = fheuint.decrypt(&client_key);
+
+```
+
+## Second Use Case: Pregenerated TFHE-rs keys
```

Comment:
I would also have a link to the full code please, it's much easier for the reader to read and have the full story than just having snippets. And ideally this full code would be tested, to avoid to have broken examples

Reply:
The reason I went with snippet is because we will have to switch between Python and Rust. We can have a simple complete code. If we want to have a full code that works, it will have lots of boilerplate code that will make calls to `os.system`. You can see the tests as an example.

I would suggest we add a link to the tests as they are complete examples.

Reply:
What you think is better, yes. From the discussion with the customer, they said the pytest was maybe a bit hard to understand so if we can have something self contained and easier to understand, great. 

Or maybe we could have a complete tutorial in `concrete/frontends/concrete-python/examples/tfhe-rs`?

Reply:
I have created the example directory, I will fill it next!